### PR TITLE
Rework the Travis configuration to use bazelisk and handle android.

### PR DESCRIPTION
**Pull request by cgruber**

### CODE REVIEW

**Comment by cgruber** [comment]
Could I get a review here please, @inez?

File: .travis.yml [hunk 1]
```diff
@@ -1,54 +1,61 @@
...
+      - adoptopenjdk/openjdk
+    casks:
+      - adoptopenjdk8
     update: true
 
 before_install:
-  - source tools/.travis.${TRAVIS_OS_NAME}
-  - wget ${BAZEL_DOWNLOAD_PREFIX}/${BAZEL_VERSION}/${BAZEL_INSTALLER}
-  - ${SHA_CMD} -c tools/${BAZEL_INSTALLER}.sha256
+  - if [ "osx" == "${TRAVIS_OS_NAME}" ]; then export PLATFORM="darwin" ; else export PLATFORM="${TRAVIS_OS_NAME}" ; fi
```
Comment (inez) [inline .travis.yml]: nit: You could combine this statement with the one below, and also intend it nicely for better reading experience.

File: .travis.yml [hunk 2]
```diff
@@ -1,54 +1,61 @@
...
+  - if [ "osx" == "${TRAVIS_OS_NAME}" ]; then export JAVA_HOME="/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home" ; fi
+  - if [ "linux" == "${TRAVIS_OS_NAME}" ]; then export JAVA_HOME="/usr/lib/jvm/java-8-openjdk-amd64" ; fi
+  - ls -FaCl ${JAVA_HOME}/bin
```
Comment (inez) [inline .travis.yml]: Is this intentional? For debugging purposes?

Comment (cgruber) [reply .travis.yml]: Yeah, this makes sure you can see the binaries, so when it fails, it's clear that, for instance, a java isn't installed in this location, etc.

File: .travis.yml (hunk 1 above)
Comment (cgruber) [reply .travis.yml]: I agree, but have had some issues with travis and spreading shell expressions across multiple lines in this yaml list, so I'm leaving it like this as a precaution.

Down the road, I will probably extract this into a sourced shell artifact.

**Pull request merged**